test/server/content-charset-test.js

```diff
@@ -0,0 +1,287 @@
+                res.end(iconv.encode(pageWithoutMetaSrc, charsetDefault));
+            });
+
+        app
+            .get('/script-with-bom', function (req, res) {
+                res.set('content-type', 'application/javascript; charset=' + charsetContentType);
+                res.end(iconv.encode(scriptSrc, charsetBOM, { addBOM: true }));
+            })
+            .get('/script-with-content-type-header', function (req, res) {
+                res.set('content-type', 'application/javascript; charset=' + charsetContentType);
+                res.end(iconv.encode(scriptSrc, charsetContentType));
+            })
+            .get('/script-with-charset-in-url', function (req, res) {
+                res.set('content-type', 'application/javascript');
+                res.end(iconv.encode(scriptSrc, charsetUrl));
+            });
+
+        app
+            .get('/other-resource/:resourceType/:charsetType', function (req, res) {
+                var contentType;
+                var src;
+
+                switch (req.params.resourceType) {
+                    case 'stylesheet':
+                        contentType = 'text/css';
+                        src         = stylesheetSrc;
+                        break;
+                    case 'manifest':
+                        contentType = 'text/cache-manifest';
+                        src         = manifestSrc;
+                        break;
+                }
+
+                switch (req.params.charsetType) {
+                    case 'bom':
+                        res.set('content-type', contentType + '; charset=' + charsetContentType);
+                        res.end(iconv.encode(src, charsetBOM, { addBOM: true }));
+                        break;
+                    case 'content-type':
+                        res.set('content-type', contentType + '; charset=' + charsetContentType);
+                        res.end(iconv.encode(src, charsetContentType));
+                        break;
+                    default:
+                        res.set('content-type', contentType);
+                        res.end(iconv.encode(src, charsetDefault));
+                }
+            });
+
+        destServer = app.listen(2000);
+
+        session                    = new Session();
+        session.getAuthCredentials = function () {
+            return null;
+        };
+        session.id                 = 'sessionId';
+
+        proxy = new Proxy(proxyHostname, proxyPort1, proxyPort2);
+        proxy.openSession('http://127.0.0.1:2000/', session);
+    });
+
+    after(function () {
+        destServer.close();
+        proxy.close();
+        requestAgent.resetKeepAliveConnections();
+    });
+
+    // Tests
+    describe('Pages', function () {
+        var charsetTestMatch  = new RegExp('<meta class="' + SHARED_CONST.SHADOW_UI_CHARSET_CLASSNAME + '" charset="([^\"]+)"', '');
+        var charsetTestFunc   = function (originUrl, expectedCharset, expectedBody, callback) {
```

Comment:
We don't use functional expressions. `charsetTestFunc` -> `testDocumentEncoding`, `callback` -> `done`